### ISSUE TITLE
Option to set Axis range

### ISSUE THREAD

**Issue by nitroSkirata** [Snoothy/UCR#170]
Hey,
It would be really nice to have an option to set the range of and output axis.

What i mean:
currently the axis go only from -50 to +50, for a total range of 100.
Id like to be al=ble to change this to be from 0 to 100 for a total range of 100.

this is important for me, since i want one axis to have a higher sensitivity, always being 120% above the original axis and not 80% below the original axis, for the first half of the travel.

**Comment by evilC** [comment]
This is incorrect. Internally, axis values are actually -32768...32767
Various GuiControls (Deadzone, Sensitivity etc) use differing scales - 
 Deadzone uses 0..100 (%)
Controls such as the "Press Value" for ButtonToAxis use a scale of -100..+100 (%)
Sensitivity uses an infinite scale: -200% or +100% or +100000% are all valid values
Nothing uses -50...+50 as far as I am aware

If you want one axis to be 180% as sensitive as another, use a Sensitivity value of 180%
This would result in a curve, if you do not want a curve and are OK with not all of the input axis mapping to a discrete output value, then check the Linear option

**Comment by nitroSkirata** [comment]
Okay,
maybe i explained my problem wrong:

I am flying a 4 engined aircraft in a flightsim and use UCR to duplicate my throlle axis, to have one for each engine.
each engine throttles from 0 to 100%, but engine 3 needs a tad more throttle then the others.

therefore i'd like the throttle to always add to the input axis, and not subtract for the first half of the travel.

Since i had no idea what the actual values were, i just use the arbitrary 0 to 100 scale in my first post.
![Untitled1](https://user-images.githubusercontent.com/9157835/81837704-79627880-9545-11ea-8d48-5ee4c28478f4.png)
![Untitled2](https://user-images.githubusercontent.com/9157835/81837705-79627880-9545-11ea-9bb4-ed5c63c941ee.png)

**Comment by evilC** [comment]
Ohh, I see what you are saying
OK, we wouldn't need to change the scale, we would just need a new sensitivity option that treated the bottom left as the origin rather than the middle

**Comment by evilC** [comment]
Try this
[AxisToAxisUnsignedSensitivity.zip](https://github.com/Snoothy/UCR/files/4623652/AxisToAxisUnsignedSensitivity.zip)
(Right click the Zip, properties, check unblock, then extract and place in the UCR.Plugins folder next to the existing UCR.Plugins.dll)
Adds a new plugin "Axis To Axis (Unsigned Sensitivity)"

**Comment by nitroSkirata** [comment]
Hey, that was a quick answer , thanks :D

though i think i miss something, as i can't set the sensitivity.
![Annotation 2020-05-13 213853](https://user-images.githubusercontent.com/9157835/81857269-50041580-9562-11ea-91b4-a1f3f2fc2f92.jpg)

**Comment by evilC** [comment]
D'oh, dunno what happened there
[AxisToAxisUnsignedSensitivity.zip](https://github.com/Snoothy/UCR/files/4625020/AxisToAxisUnsignedSensitivity.zip)

**Comment by nitroSkirata** [comment]
Thanks, just trying it out, and it works like a charm :D

the only further suggestion i'd have is to add a invert sensitivity option (change from wich end of the axis the sensitivity is applied)

for me currently the sensitivity seems to be applied from the high end (when is input is at maximum, the output equals the input).
![Untitled1](https://user-images.githubusercontent.com/9157835/81924576-951a5d00-95df-11ea-8bba-bb8f9ead2c9c.png)

**Comment by evilC** [comment]
I had a quick stab at it, but cannot seem to wrap my head around the maths  

The existing code does this (`value` is a signed 16-bit integer -32768...32767)
1. Add 32768 to `value` (Shifts scale to 0...65535)
1. Multiply `value` by sensitivityFactor (eg for 150%, multiply by 1.5)
1. Subtract `32768` from value
1. Cap `value` to between -32768 and 32767

so eg for a mid-point value of 0:
1. 0 + 32768 = 32768
1. 32768 * 1.5 = 49152
1. 49152 - 32768 = 16384
1. Cap to -32768...32767 = leaves at 16384

for a min value of -32768
1. -32768 + 32768 = 0
1. 0 * 1.5 = 0
1. 0 - 32768 = -32768
1. Cap does nothing

for a max value of 32767:
1. 32767 + 32768 = 65535
1. 65535 * 1.5 = 98302.5
1. 98302.5 - 32768 = 65534.5
1. Cap to -32768...32767 = 32767

Can you work out the maths for what you want?

**Comment by nitroSkirata** [comment]
tbh, i have no idea how the maths work, but i can explain how i went around it:

1.In UCR I enable the invert setting
2. i added the sensitivity multiplier (104%)
3. i inverted the axis in the game

I hope this helps you
